src/main/scala/org/broadinstitute/dsde/workbench/sam/db/DbReference.scala

```diff
@@ -42,38 +49,72 @@ object DbReference extends LazyLogging {
     }
   }
 
-  def init(liquibaseConfig: LiquibaseConfig, dbName: DatabaseName): DbReference = {
+  def init(liquibaseConfig: LiquibaseConfig, dbName: DatabaseName, dbExecutionContext: ExecutionContext): DbReference = {
     DBs.setup(dbName.name)
     DBs.loadGlobalSettings()
     if (liquibaseConfig.initWithLiquibase) {
-      initWithLiquibase(liquibaseConfig)
+      initWithLiquibase(liquibaseConfig, dbName)
     }
 
-    DbReference(dbName)
+    DbReference(dbName, dbExecutionContext)
   }
 
-  def resource(liquibaseConfig: LiquibaseConfig, dbName: DatabaseName): Resource[IO, DbReference] = Resource.make(
-    IO(init(liquibaseConfig, dbName))
-  )(_ => IO(DBs.close(dbName.name)))
+  def resource(liquibaseConfig: LiquibaseConfig, dbName: DatabaseName): Resource[IO, DbReference] = {
+    for {
+      dbExecutionContext <- ExecutionContexts.fixedThreadPool[IO](DBs.config.getInt(s"db.${dbName.name.name}.poolMaxSize"))
```

Comment:
note this change moves the thread pool associated to a db connection internal to the db ref.